src/Mvc/Mvc.ApiExplorer/src/EndpointMetadataApiDescriptionProvider.cs

```diff
@@ -294,6 +306,73 @@ private ApiDescription CreateApiDescription(RouteEndpoint routeEndpoint, string
             }
         }
 
+        private static List<ApiResponseType> ReadResponseMetadata(
+            IReadOnlyList<IProducesResponseTypeMetadata> responseMetadataAttributes,
+            Type? type,
+            Type defaultErrorType)
+        {
+            var results = new Dictionary<int, ApiResponseType>();
+
+            // Get the content type that the action explicitly set to support.
+            // Walk through all 'filter' attributes in order, and allow each one to see or override
+            // the results of the previous ones. This is similar to the execution path for content-negotiation.
+            if (responseMetadataAttributes != null)
+            {
+                foreach (var metadataAttribute in responseMetadataAttributes)
+                {
+                    var statusCode = metadataAttribute.StatusCode;
+
+                    var apiResponseType = new ApiResponseType
+                    {
+                        Type = metadataAttribute.Type,
+                        StatusCode = statusCode,
+                        IsDefaultResponse = metadataAttribute is IApiDefaultResponseMetadataProvider,
```

Comment:
Do we need to support this still? We should be using all new metadata only.

Reply:
Merp -- fixed.